src/TerminalFighter/Title/Title.h

```diff
@@ -0,0 +1,17 @@
+#include <string>
+#include <SDL_ttf.h>
+
+class Title {
+    std::string title_str;
+    SDL_Color color;
+    int x_pos;
+    int y_pos;
+    int width;
+    int height;
+    TTF_Font* font;
+public:
+    Title() : title_str(), color(), x_pos(0), y_pos(0), width(0), height(0), font() {};
```

Comment:
Only the prototype should go in the header, this is an implementation. I kinda doesn't make sense to this class with default parameters, so IMO we should just forbid the default constructor altogether.